base/sparse/sparsematrix.jl

```diff
@@ -763,6 +763,11 @@ function sparse(B::Bidiagonal)
     return sparse([1:m;1:m-1],[1:m;2:m],[B.dv;B.ev], Int(m), Int(m)) # upper bidiagonal
 end
 
+function sparse(D::Diagonal{T}) where T
+    m = length(D.diag)
+    return SparseMatrixCSC(m, m, [1:m; m+1], [1:m;], convert(Vector{T}, D.diag))
```

Comment:
Might the `convert(Vector{T}, D.diag)` produce unexpected aliasing if `D.diag isa `Vector{T}`?

Reply:
Yes, but that was intentional from my side :smile:

Reply:
Cheers :). Do other `sparse` methods alias their inputs?

Reply:
No, because the can't really do that; there are no other arrays that have all their nzvals in one ordered vector like `Diagonal` does. Perhaps best to make a copy here?

Reply:
Perhaps copying here as well would be for the best then, yes :). Thoughts?